document request handling flow (#71)
kubernetes-sigs/oci-proxy: cmd/archeio/docs/request-handling.md

# Request Handling

Requests to archeio follows the following flow:

1. If it's a request for `/`: redirect to our wiki page about the project
2. If it's not a request for `/` and does not start with `/v2/`: 404 error
3. For registry API requests, all of which start with `/v2/`:
  - If it's not a blob request: redirect to GCR

>   - If it's not a known AWS IP: redirect to GCR

Reviewer: ```suggestion
  - If it's not a known AWS IP: redirect to [k8s.gcr.io](https://k8s.gcr.io)
```

>   -  If it's a known AWS IP AND HEAD reuqest for the layer succeeeds in S3: redirect to S3

Reviewer 2: typo `reuqest`

>   -  If it's a known AWS IP AND HEAD reuqest for the layer succeeeds in S3: redirect to S3

Reviewer: ```suggestion
  -  If it's a known AWS IP AND HEAD request for the layer succeeds in S3: redirect to the S3 bucket
```

>   -  If it's a known AWS IP AND HEAD fails: redirect to GCR

Reviewer: ```suggestion
  -  If it's a known AWS IP AND HEAD fails: redirect to [k8s.gcr.io](https://k8s.gcr.io)
```
